src/agent/debuglet.ts

```diff
@@ -322,6 +349,32 @@ export class Debuglet extends EventEmitter {
   }
 
   /**
+   * isReady returns a promise that only resolved if the last breakpoint update
+   * happend within a duration (PROMISE_RESOLVE_CUT_OFF_IN_MILLISECONDS). This
+   * feature is mainly used in Google Cloud Function (GCF), as it is a
+   * serverless environment and we wanted to make sure debug agent always
+   * captures the snapshots.
+   */
+  async isReady(): Promise<void> {
+    if (Date.now() < this.breakpointFetchedTimestamp +
+            PROMISE_RESOLVE_CUT_OFF_IN_MILLISECONDS) {
+      return Promise.resolve();
+    } else {
+      if (this.breakpointFetched) {
+        return this.breakpointFetched;
+      }
+      this.debuggeeRegistered.then(() => {
+        this.breakpointFetched = new Promise<void>((resolve) => {
```

Comment:
This looks like `CachedPromise` in #341, since this Promise also goes stale. Is it possible to use that abstraction for `breakpointFetched` (and also `debuggeeRegistered`)? For the latter, since it never goes stale you could use a value of `Infinity` for time until stale.

Reply:
So in #341, `CachedPromise` would also include a time variable. I have thought to use that abstraction here as well as we have two promises (`breakpointFetched`, `debuggeeRegistered`). But checking back their behaviors, `debuggeeRegistered` will be resolved after debuggee has been successfully registered and that's done. However, for `breakpointFetched`, when it has been resolved, it needs to be set to null so that next time when GCF calls `isReady', it can create a new promise instead of using the stale one. The logics would be more clean if not using abstraction in this case.